### DISCUSSION TITLE
nmap building site-packages using system installed python 2.7

### DISCUSSION THREAD

**Discussion by ghost** [Homebrew/discussions#2140] [Everyday usage]
Is this expected behavior?
If you have a command that would clue me in please post, thank you.
I tried `brew deps --tree nmap` but that didn't show what version of python nmap requires
I uninstalled and reinstalled nmap and had the same result.

```
$ which python
/Users/john/.pyenv/shims/python

$ python -V
Python 3.9.7

$ find /usr -name 'site-packages' | grep "nmap"
find: /usr/sbin/authserver: Permission denied
/usr/local/Cellar/nmap/7.92/lib/python2.7/site-packages
```

**Comment by SMillerDev** [comment]
The code at https://github.com/Homebrew/homebrew-core/blob/e3d2f98c6f220f5d2c6720111cde60431b7fec74/Formula/nmap.rb

Doesn't do anything related to python. It might be accidentally picking up system python in the install though. In that case we probably want to find a way to disable it.

**Reply by ghost** [reply]
Beyond my knowledge, but thought I should mention it. I didn't know if it used python or not. There is a pypi package called python-nmap that I think uses the nmap executable. I don't know if it's installed in the package or looks for an existing installation.

**Reply by ghost** [reply]
Just wanted to add that there is an `ndiff.py` file inside the site-packages directory. Does `ndiff` still require python 2.7? If so then this is to be expected.

```
/usr/local/Cellar/nmap/7.92/.brew
/usr/local/Cellar/nmap/7.92/bin
/usr/local/Cellar/nmap/7.92/CHANGELOG
/usr/local/Cellar/nmap/7.92/INSTALL_RECEIPT.json
/usr/local/Cellar/nmap/7.92/lib
/usr/local/Cellar/nmap/7.92/lib/python2.7
/usr/local/Cellar/nmap/7.92/lib/python2.7/site-packages
/usr/local/Cellar/nmap/7.92/lib/python2.7/site-packages/ndiff.py
/usr/local/Cellar/nmap/7.92/LICENSE
/usr/local/Cellar/nmap/7.92/README.md
/usr/local/Cellar/nmap/7.92/share
```
